eventhubs/data-plane/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/EventHubClientImpl.java

```diff
@@ -54,6 +53,9 @@ private EventHubClientImpl(final ConnectionStringBuilder connectionString, final
 
         this.eventHubName = connectionString.getEventHubName();
         this.senderCreateSync = new Object();
+        this.underlyingFactory = null;
+        this.sender = null;
+        this.timer = null;
```

Comment:
I saw nowhere the sender is initialzied in the class and all of three are "private volatile". It should  be null by default.  @JamesBirdsall  Do you know how the sender is initialized here.

Reply:
It is null by default - when it is defined in the class it is not set to anything, so it will be null. Setting it to null in the constructor doesn't appear to do anything.

Reply:
Yes, that is right. I would suppress this error since adding null values to these values is a dummy operation but the SpotBugs error message will be gone.  Note: adding null value on the constructor will not effect the initialization of the MessageSender sender. The sender is lazy initialized at run time when the EventHub Client sends the message to service at first time(the method initialize the sender when it is null).   For the other two variable (timer and underlyingFactory ), they are initialized when EventHubClient create the connection with service. So initializing null to these two variables in the EventHubClient constructor will not affect the initialization of them inside of EventHubClient.create() method.